test/lib/smart-survey-api.test.js

```diff
@@ -1,107 +1,57 @@
-
-        // reproxyrequire because the stubSmartSurvey now has a stub
-        smartSurveyClientAPI = proxyquire('../../lib/smart-survey-api', {
-          'smartsurvey-client': stubSmartSurvey
-        });
-
-        smartSurveyClientAPI.createClient.call(context, '1', '2');
-        stubSmartSurvey.should.have.been.called;
-    });
-    it('should call the callback with an error response when we receive an error from the api', function() {
-
-      context = {
-        createClient: sinon.stub(),
-        client: {
-          getResponses: sinon.stub()
-        }
-      };
-      const callback = sinon.stub();
-
-      // Causes the stub to call the argument at the index as a callback function.
-      // stub.callsArg(2); stub calls the 3rd argument as a callback.
-      // callsArgWith like callsArg, but with arguments to pass to the callback.
-      context.client.getResponses.callsArgWith(2, 'error message', 1);
-      smartSurveyClientAPI.getData.call(context, 'a', 'b', undefined, callback);
-      callback.should.have.been.calledWith('error message', 1);
-    });
-    it('should call the callback without an error when we receive success from the api', function() {
-
-      context = {
-        createClient: sinon.stub(),
-        client: {
-          getResponses: sinon.stub()
-        }
-      };
-      const callback = sinon.stub();
-
-      context.client.getResponses.callsArgWith(2, null, 'good');
-      smartSurveyClientAPI.getData.call(context, 'a', 'b', undefined, callback);
-      callback.should.have.been.calledWith(null, 'good');
-    });
+// 2nd argument: An object with the key: module name to proxy & the value the object above
+// So this means smartSurveyStub will now point to this function, i.e. a stub
+const smartSurveyAPI = proxyquire('../../lib/smart-survey-api', { 'smartsurvey-client': smartSurveyStub});
+
+describe('smartSurveyAPI', () => {
+  describe('getData()', () => {
+    it('is a function', () => (typeof smartSurveyAPI.getData).should.equal('function'));
+    it('takes 3 arguments', () => (smartSurveyAPI.getData).should.have.lengthOf(3));
+
+    describe('when HAPPYPATH', () => {
+      let result;
+      const response = {foo: 'bar'};
+
+      before(() => {
+        getResponsesStub
+          .withArgs('survey1', sinon.match.any)
+          .yields(null, response);
+        result = smartSurveyAPI.getData('mytoken', 'mytokensecret', 'survey1');
+      });
+
+      after(() => {
+        getResponsesStub.reset();
+      });
+
+      it('returns a Promise', () => result.should.be.a('Promise'));
+      it('resolves', () => result.should.eventually.be.fulfilled);
+      it('that resolves to the response', () => result.should.eventually.equal(response));
   });
-
-  describe('parseResponse', function() {
-    it('should return an error message if there is an error', function() {
-      parseResponse('error', null).should.equal('error');
+    describe('when NOT HAPPYPATH', () => {
```

Comment:
'when SmartSurvey returns an error' ??

I'd consider the HAPPYPATH and NOT HAPPYPATH as a form of tech-debt. As there can be lots of different non-happy paths and you should document which scenario you are testing.